internal/controllers/addon/controller.go

```diff
@@ -287,6 +295,10 @@ func (r *AddonReconciler) Reconcile(
 	}
 	errors := r.syncWithExternalAPIs(ctx, logger, addon)
 
+	if errors.ErrorOrNil() == nil {
```

Comment:
If I'm interpreting this right, this will report a reconcile error if the error list in empty. Is this expected?

Reply:
Nope, good catch.